components/scream/cmake/machine-files/perlmutter.cmake

```diff
@@ -1,11 +1,25 @@
 # Load all kokkos settings from Ekat's mach file
 set (EKAT_MACH_FILES_PATH ${CMAKE_CURRENT_LIST_DIR}/../../../../externals/ekat/cmake/machine-files)
-include (${EKAT_MACH_FILES_PATH}/kokkos/nvidia-a100.cmake)
-include (${EKAT_MACH_FILES_PATH}/kokkos/cuda.cmake)
+IF (USE_CUDA OR KOKKOS_ENABLE_CUDA)
+  include (${EKAT_MACH_FILES_PATH}/kokkos/nvidia-a100.cmake)
+  include (${EKAT_MACH_FILES_PATH}/kokkos/cuda.cmake)
+else()  
+  include (${EKAT_MACH_FILES_PATH}/kokkos/amd-zen2.cmake)
+  #include (${EKAT_MACH_FILES_PATH}/kokkos/openmp.cmake)
+  include (${EKAT_MACH_FILES_PATH}/kokkos/serial.cmake)
+endif()
 
 #option(Kokkos_ARCH_AMPERE80 "" ON)
 set(CMAKE_CXX_FLAGS "-DTHRUST_IGNORE_CUB_VERSION_CHECK" CACHE STRING "" FORCE)
 
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
```

Comment:
This is another thing I'm not sure it's portable. The compiler ID is set by cmake only after the language has been enabled (e.g., during the call to `project()`), so I suspect this cannot be set by the time the mach file is parsed...

Again, it might be this works, but I am not confident it's portable. I should check.